crates/block-producer/src/transaction_skeleton.rs

```diff
@@ -80,6 +84,13 @@ impl TransactionSkeleton {
     pub fn signature_entries(&self) -> Vec<SignatureEntry> {
         let mut entries: HashMap<[u8; 32], SignatureEntry> = Default::default();
         for (index, input) in self.inputs.iter().enumerate() {
+            // Skip withdrawal lock witness args
+            if let Some(witness_args) = self.witnesses().get(index) {
```

Comment:
Why do we have signed witnesses?

Reply:
These are not signed withnesses, they are ```UnlockWithdrawalWitness``` and use to revert withdrawal cells back to custodian cells,